tgui/packages/tgui/interfaces/Manufacturer/index.tsx

```diff
@@ -0,0 +1,379 @@
+      Card Owner: {data.card_owner}<br />
+      Current Balance: {data.card_balance}⪽
+      <Button icon="subtract" onClick={() => act("card", { "remove": true })}>Remove Card</Button>
+    </Box>
+  );
+};
+
+const is_set = (bits, bit) => { return bits & (1 << bit); };
+
+export const CollapsibleWireMenu = (props, context) => {
+  const { act, data } = useBackend<ManufacturerData>(context);
+  let wireContent = [];
+  let i = 0;
+  wireContent.push(
+    <>
+      <Flex style={{ "align-items": "center" }}>
+        <Flex.Item style={{ "width": "45%" }}>
+          Panel {data.panel_open ? "open" : "closed"}.
+        </Flex.Item>
+        <Flex.Item grow>
+          <Button
+            icon={data.panel_open ? "warning" : "circle"}
+            color={data.panel_open ? "orange" : "green"}
+            onClick={() => act("toggle_panel")}
+          >
+            Screws {data.panel_open ? "Loose" : "Secure"}
+          </Button>
+        </Flex.Item>
+      </Flex>
+      <Divider />
+    </>
+  );
+  if (data.panel_open)
+  {
+    for (let wire in data.wires)
+    {
+      let cut = is_set(data.wire_bitflags, i);
+      i++;
+      wireContent.push(
+        <Flex textColor={data.wires[wire]} px={2}>
+          <Flex.Item grow bold>
+            {wire}
+          </Flex.Item>
+          <Flex.Item mr="5%">
+            <Button
+              ml="10%"
+              my="1%"
+              content="Pulse"
+              onClick={() => act('wire', { action: "pulse", wire: i })}
+            />
+          </Flex.Item>
+          <Flex.Item>
+            <Button
+              m="1%"
+              content={cut ? "Cut" : "Mend"}
+              onClick={() => act("wire", { action: (cut ? "mend" : "cut"), wire: i })}
+            />
+          </Flex.Item>
+        </Flex>
+      );
+    }
+  }
+  return (
+    <Collapsible
+      mb="1%"
+      title="Maintenence Panel"
+      open={data.panel_open}
+    >
+      <Box backgroundColor={backgroundPop} p={1}>
+        {wireContent}
+        <Divider />
+        <LabeledList>
+          <LabeledList.Item
+            label="Electrification Risk">
+            {data.indicators.electrified ? "High" : "None"}
+          </LabeledList.Item>
+          <LabeledList.Item
+            label="System Stability">
+            {data.indicators.malfunctioning ? "100%" : "0%"}
+          </LabeledList.Item>
+          <LabeledList.Item
+            label="Warranty">
+            {data.indicators.hacked ? "Void" : "Valid"}
+          </LabeledList.Item>
+          <LabeledList.Item
+            label="Power">
+            {data.indicators.hasPower ? "Sufficient" : "Insufficient"}
+          </LabeledList.Item>
+        </LabeledList>
+      </Box>
+    </Collapsible>
+  );
+};
+
+export const Manufacturer = (_, context) => {
+  const { act, data } = useBackend<ManufacturerData>(context);
+  const [repeat, toggleRepeatVar] = useSharedState(context, "repeat", data.repeat);
```

Comment:
Why not just pull this from data?

Reply:
going to see about moving act, data through the chain of.. functions? i guess they're functions. i imagine useBackend is expensive

Reply:
I mean you're doing it directly above in this component, so...

Reply:
we now pull from data in Manufacturer for other components yay